snow/consensus/snowball/flat.go

```diff
@@ -28,17 +28,7 @@ type Flat struct {
 
 func (f *Flat) RecordPoll(votes bag.Bag[ids.ID]) bool {
 	pollMode, numVotes := votes.Mode()
-	switch {
-	// AlphaConfidence is guaranteed to be >= AlphaPreference, so we must check
-	// if the poll had enough votes to increase the confidence first.
-	case numVotes >= f.params.AlphaConfidence:
-		f.RecordSuccessfulPoll(pollMode)
-		return true
-	case numVotes >= f.params.AlphaPreference:
-		f.RecordPollPreference(pollMode)
-		return true
-	default:
-		f.RecordUnsuccessfulPoll()
-		return false
-	}
+	f.Nnary.RecordPoll(numVotes, pollMode)
```

Comment:
I don't think this guarantees the invariant on `RecordPoll` that `numVotes >= AlphaPreference`. (I don't think this is a bug fwiw, I think that the invariants on `RecordPoll` about expecting at least `AlphaPreference` is not needed.

Reply:
Ya `RecordPoll` implementations no longer actually assume this, but will update since it's a documented invariant anyways.

Could go the other way and remove `RecordUnsuccessfulPoll`, but we currently use `RecordUnsuccessfulPoll` to handle a consensus "falter" as well, which we could hardcode to `RecordPoll(0, choice)`, but it seems cleaner to have a separate function as it currently is.